ci/src/revisions/git_lib.py

```diff
@@ -0,0 +1,163 @@
+
+    result = subprocess.run(["git", "status", "--porcelain"], stdout=subprocess.PIPE, text=True, check=True)
+    if result.stdout.strip():
+        logger.warn("Uncommited work:\n%s", result.stdout.strip())
+        logger.error("Found uncommited work! Commit and then proceed.")
+        exit(2)
+
+    if subprocess.call(["git", "checkout", branch_to_checkout], cwd=repo_root) == 0:
+        # The branch already exists, update the existing MR
+        logger.info("Found an already existing target branch")
+    else:
+        subprocess.check_call(["git", "checkout", "-b", branch_to_checkout], cwd=repo_root)
+    subprocess.check_call(["git", "reset", "--hard", f"origin/{main_branch}"], cwd=repo_root)
+
+
+def commit_and_create_pr(
+    repo: str, repo_root: pathlib.Path, branch: str, check_for_updates_in_paths: List[str], logger: logging.Logger
+):
+    git_modified_files = subprocess.check_output(["git", "ls-files", "--modified", "--others"], cwd=repo_root).decode(
+        "utf8"
+    )
+
+    paths_to_add = [path for path in check_for_updates_in_paths if path in git_modified_files]
+
+    if len(paths_to_add) > 0:
+        logger.info("Creating/updating a MR that updates the saved NNS subnet revision")
+        cmd = ["git", "add"] + paths_to_add
+        logger.info("Running command '%s'", " ".join(cmd))
+        subprocess.check_call(cmd, cwd=repo_root)
+        cmd = [
+            "git",
+            "-c",
+            "user.name=CI Automation",
+            "-c",
+            "user.email=infra+github-automation@dfinity.org",
+            "commit",
+            "-m",
+            "chore: Update Mainnet IC revisions file",
+        ] + paths_to_add
+        logger.info("Running command '%s'", " ".join(cmd))
+        subprocess.check_call(
+            cmd,
+            cwd=repo_root,
+        )
+        subprocess.check_call(["git", "push", "origin", branch, "-f"], cwd=repo_root)
+
+        if not subprocess.check_output(
+            ["gh", "pr", "list", "--head", branch, "--repo", repo],
+            cwd=repo_root,
+        ).decode("utf8"):
+            subprocess.check_call(
+                [
+                    "gh",
+                    "pr",
+                    "create",
+                    "--head",
+                    branch,
+                    "--repo",
+                    repo,
+                    "--fill",
+                ],
+                cwd=repo_root,
+            )
+
+
+def run_sync_main_branch_and_checkout_branch(args, logger):
+    repo_root = args.repo_root
+    branch = args.branch
+    main_branch = args.main_branch
+    sync_main_branch_and_checkout_branch(repo_root, main_branch, branch, logger)
+
+
+def run_commit_and_create_pr(args, logger):
+    repo = args.repo
+    repo_root = args.repo_root
+    branch = args.branch
+    files = args.files
+    commit_and_create_pr(repo, repo_root, branch, files, logger)
+
+
+if __name__ == "__main__":
```

Comment:
Best practice is that this section stays pretty empty and that all code is moved into other functions.